# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTypes.h
# Commit: 40031ad88dbd
# Full Hash: 40031ad88dbd9af3296493ae4bf501c158cb0ca2
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-11-03 21:39:33
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
#   All these places require short fixes to use our new scheme for getting
# ==============================================================================

diff -r cee7eaa8c402 -r 40031ad88dbd js/src/wasm/WasmTypes.h
--- a/js/src/wasm/WasmTypes.h	Tue Nov 03 10:25:14 2020 +0000
+++ b/js/src/wasm/WasmTypes.h	Tue Nov 03 10:28:10 2020 +0000
@@ -3267,7 +3267,6 @@
 
   uint8_t* rawCaller() const { return callerFP_; }
   TlsData* tls() const { return tls_; }
-  Instance* instance() const { return tls()->instance; }
 
   Frame* wasmCaller() const {
     MOZ_ASSERT(!callerIsExitOrJitEntryFP());
@@ -3427,7 +3426,7 @@
   static DebugFrame* from(Frame* fp);
   Frame& frame() { return frame_; }
   uint32_t funcIndex() const { return funcIndex_; }
-  Instance* instance() const { return frame_.instance(); }
+  Instance* instance() const;
   GlobalObject* global() const;
   bool hasGlobal(const GlobalObject* global) const;
   JSObject* environmentChain() const;
